src/_gettsim_tests/_policy_test_utils.py

```diff
@@ -1,158 +1,111 @@
-            test
-            for test in self.test_data
-            if (test_name is None or test.test_name == test_name)
-            and (date is None or test.date == date)
-        ]
+    root = TEST_DATA_DIR / policy_name
 
-        return PolicyTestSet(self.policy_name, filtered_test_data)
+    out = []
 
+    for path_of_test_file in root.glob("**/*.yaml"):
+        if _is_skipped(path_of_test_file):
+            continue
 
-class PolicyTestData:
-    def __init__(  # noqa: PLR0913
-        self,
-        policy_name: str,
-        test_file: Path,
-        test_name: str,
-        date: str,
-        inputs_provided: _ValueDict,
-        inputs_assumed: _ValueDict,
-        outputs: _ValueDict,
-    ):
-        self.policy_name = policy_name
-        self.test_file = test_file
-        self.test_name = test_name
-        self.date = _parse_date(date)
-        self._inputs_provided = inputs_provided
-        self._inputs_assumed = inputs_assumed
-        self._outputs = outputs
+        with path_of_test_file.open("r", encoding="utf-8") as file:
+            raw_test_data: NestedDataDict = yaml.safe_load(file)
 
-    @property
-    def input_df(self) -> pd.DataFrame:
-        return pd.DataFrame.from_dict(
-            {**self._inputs_provided, **self._inputs_assumed}
-        ).reset_index(drop=True)
+        out.extend(
+            _get_policy_tests_from_raw_test_data(
+                raw_test_data=raw_test_data,
+                path_of_test_file=path_of_test_file,
+            )
+        )
 
-    @property
-    def output_df(self) -> pd.DataFrame:
-        return pd.DataFrame.from_dict(self._outputs).reset_index(drop=True)
+    return out
 
-    def __repr__(self) -> str:
-        return (
-            f"PolicyTestData({self.policy_name}, {self.test_file.name}, "
-            f"{self.test_name})"
-        )
 
-    def __str__(self) -> str:
-        relative_path = self.test_file.relative_to(TEST_DATA_DIR)
-        backslash = "\\"
-        return f"{str(relative_path).replace(backslash, '/')}"
+def _is_skipped(test_file: Path) -> bool:
+    return "skip" in test_file.stem or "skip" in test_file.parent.name
 
 
-def load_policy_test_data(policy_name: str) -> PolicyTestSet:
-    from _gettsim_tests import TEST_DATA_DIR
+def _get_policy_tests_from_raw_test_data(
+    raw_test_data: NestedDataDict, path_of_test_file: Path
+) -> list[PolicyTest]:
+    """Get a list of PolicyTest objects from raw test data.
 
-    root = TEST_DATA_DIR / policy_name
+    Args:
+        raw_test_data: The raw test data.
 
+    Returns:
+        A list of PolicyTest objects.
+    """
     out = []
 
-    for test_file in root.glob("**/*.yaml"):
-        if _is_skipped(test_file):
-            continue
-
-        with test_file.open("r", encoding="utf-8") as file:
-            test_data: dict[str, dict] = yaml.safe_load(file)
+    inputs: NestedDataDict = raw_test_data.get("inputs", {})
+    input_tree: NestedDataDict = merge_trees(
```

Comment:
```suggestion
    input_tree: NestedDataDict = optree.treee_map(
        np.array,
        merge_trees(
            inputs.get("provided", {}), inputs.get("assumed", {}
        ),
    )
```

Like this? (untested)

Reply:
I did it like this:

```python
    input_tree: NestedDataDict = flatten_dict.unflatten(
        {
            k: np.array(v)
            for k, v in flatten_dict.flatten(
                merge_trees(inputs.get("provided", {}), inputs.get("assumed", {}))
            ).items()
        }
    )
```

I didn't manage to register lists as nodes for optree, I think one has to create a custom data object to do that. (But didn't spend too much time on this.)